tests/gen_test.py

```diff
@@ -3,37 +3,52 @@
 # (C) 2021 COVESA
 # Test code for code generator part of VSC tools
 # ----------------------------------------------------------------------------
-
-import pytest
-
 # This is maybe not ideal way but seems efficient enough
-from pathlib import Path
-import sys
-proj_root = Path(__file__).parents[1]
-sys.path.append(str(proj_root) + "/model")
+from vsc.model import vsc_ast, vsc_parser, vsc_generator
+import os, pathlib
 
-import vsc_parser, vsc_generator
+TestPath = os.path.dirname(os.path.realpath(__file__))
 
 def test_x():
     assert 1 == 1
 
-def test_gen():
-    # The files named 'input', 'template' and 'result' are in the tests directory
-    ast = vsc_parser.get_ast_from_file('input')
 
-    with open("template", "r") as templatefile:
-        generated = vsc_generator._gen_with_text_template(ast, templatefile.read())
+def test_gen(tmp_path):
+    # The files named 'input.yaml', 'template' and 'result' are in the tests directory
+    ast_root = vsc_parser.get_ast_from_file(os.path.join(TestPath,'input_old.yaml'))
```

Comment:
Is this a test to support the "old" syntax (without explicit "service"). Do we intend to support both or is this just a temporary test?

Reply:
We are in the process of migration and I've kept both. But, in the end, only one syntax will be left.

Reply:
@erikbosch I have reverted most of the changes I've made with "syntax". Some minor things are still there, like `input` instead of `in`, etc., but it is not avoidable.

Reply:
@erikbosch Honestly it's all a bit of a mess at the moment!  While merging and updating some of my code on top of this I noticed we had a mix of proposed syntax changes from @miketsukerman, a few from myself, the templates, the specification, different implementations in progress, unit-tests, examples files in VSC repo - all had various levels of differences.  To get a stable situation I have also rolled back a lot to a syntax that I don't feel is the final/best one.
But I'd rather we start from there and change incrementally.  From now on, let's try to make sure all PRs will modify parts in sync. (This is still Draft PR of course, but nonetheless).

Reply:
@gunnarx let's call it `refactoring` :)